src/address.rs

```diff
@@ -0,0 +1,139 @@
+use anyhow::bail;
+use fvm_shared::address::{Address, Network};
+use fvm_shared::ActorID;
+
+// '0x' + 20bytes
+const ETH_ADDRESS_LENGTH: usize = 42;
+const EAM_NAMESPACE: ActorID = 10;
+
+fn check_address_prefix(s: &str, n: Network) -> bool {
+    if s.len() < 2 {
+        return false;
+    }
+
+    match n {
+        Network::Mainnet => s[0..1].eq("f") || s[0..2].eq("0x"),
+        Network::Testnet => s[0..1].eq("t") || s[0..2].eq("0x"),
+    }
+}
+
+fn is_eth_address(s: &str) -> bool {
+    s.len() > 2 && s[0..2].eq("0x")
+}
```

Comment:
```suggestion
fn is_eth_address(s: &str) -> bool {
    s.starts_with("0x")
        && s.len() == ETH_ADDRESS_LENGTH
        && s.chars().skip(2).all(|c| c.is_ascii_hexdigit())
}
```

Reply:
Reverted this because it makes the error handling worse, since it only returns a bool. Used these tricks in the existing checks, and added the other checks directly in parse_address

Reply:
You could use `ensure!` but no biggie. Thanks for the PR!